node/core/parachains-inherent/src/lib.rs

```diff
@@ -119,18 +120,28 @@ impl ParachainsInherentDataProvider {
 			},
 		};
 
-		Ok(Self { inherent_data })
+		Ok(inherent_data)
 	}
 }
 
 #[async_trait::async_trait]
-impl sp_inherents::InherentDataProvider for ParachainsInherentDataProvider {
-	fn provide_inherent_data(
+impl<C: sp_blockchain::HeaderBackend<Block>> sp_inherents::InherentDataProvider
+	for ParachainsInherentDataProvider<C>
+{
+	async fn provide_inherent_data(
 		&self,
 		dst_inherent_data: &mut sp_inherents::InherentData,
 	) -> Result<(), sp_inherents::Error> {
+		let inherent_data = ParachainsInherentDataProvider::create(
+			self.client.clone(),
+			self.overseer.clone(),
+			self.parent,
+		)
+		.await
+		.map_err(|e| sp_inherents::Error::Application(Box::new(e)))?;
```

Comment:
```suggestion
		let inherent_data = self.create()
		.await
		.map_err(|e| sp_inherents::Error::Application(Box::new(e)))?;
```

Reply:
Revert as this would need `self`to be `mut` which in the end would need `self` in `InherentDataProvider` to be mut which feels wrong to change for this one case.